sphinx/ext/autosummary/generate.py

```diff
@@ -320,11 +320,34 @@ def generate_autosummary_content(name: str, obj: Any, parent: Any,
         ns['members'] = dir(obj)
         ns['inherited_members'] = \
             set(dir(obj)) - set(obj.__dict__.keys())
+
         ns['methods'], ns['all_methods'] = \
             _get_members(doc, app, obj, {'method'}, include_public={'__init__'})
         ns['attributes'], ns['all_attributes'] = \
             _get_members(doc, app, obj, {'attribute', 'property'})
 
+        method_string = [m.split(".")[-1] for m in ns['methods']]
```

Comment:
Perhaps it's better to have use `ns['all_methods']` since otherwise you'll only include public ones.

Reply:
@satr-cowi Have you addressed this issue?

Reply:
I think my preference is for ns['inherited_methods'] would be to only return the public methods (and therefore be consistent with ns['methods'])

We could introduce ns['all_inherited_methods'] as well, but the documentation doesn't include anything about ns['all_methods'] at the moment so we should probably include that too if we do this.

Reply:
Personally, I want us to be consistent without breaking previous changes.

AFAIU the current behaviour is "ns[C] returns public members for category C" and "ns[all_C] returns all members for category C (including private ones)". I don't mind not documenting "all_*" for now, but I would like the documentation for ns["members"] & co. to be updated then. Currently it says (for "members"):

> List containing names of all members of the module or class. Only available
  for modules and classes.

In addition, your "inherited_qualnames" documentation tells me about: 

>    * ``members`` returns a list with many built in methods/attributes and
     ``['__init__', '_i_am_private', 'do_something', 'do_something_else',
     'foo_attr']``

So now your members also include private members.

Reply:
Current behaviour (as in the example) is:

- methods and attributes returns just public methods/attributes
- members and inherited members return all members including private
- all_methods and all_attributes returns all methods/attributes including private, but these are not documented so most users won't know they exist

I've updated the "members" documentation to make it clear that this and "inherited members" do include private members. It was already clear that methods/attributes were only public.

The proposed additions are just that:

- inherited_methods and inherited_attributes returns just public inherited methods/attributes
- inherited_qualnames is the same as inherited_members (includes private), but just with the fully qualified names

Think this is pretty clear from the documentation now